File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <stdio.h>
#include <math.h>

bool is_armstrong_number(int candidate){
    int aux = candidate;
    int length = 0;
    int flag = 1;
    int armstrong = 0;

    do{
        aux = aux/10;
        length++;
        if(aux < 1)
            flag = 0;
    }while(flag);

    aux = candidate;
    for(int i=0; i<length; i++){
        armstrong += pow(aux%10, length);
        aux = aux/10;
    }

    if(armstrong == candidate)
        return true;
    else
        return false;
}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
import java.io.*;
import java.lang.Math;

class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int exp = getNumLength(numberToCheck);
        return numberToCheck == getArmstrongNumber(numberToCheck, exp);
    }

    int getNumLength(int num){
        int ret = 0;
        double aux = num;
        while(aux > 1){
            aux = aux/10;
            ret++;
        }
        return ret;
    }

    int getArmstrongNumber(int num, int exp){
        int arm = 0;
        double aux = num;
        while(aux > 1){
            arm += Math.pow((int)(aux%10), exp);
            aux = aux/10;
        }
        return arm;
    }

}
